subprojects/messaging/src/main/java/org/gradle/internal/remote/ObjectConnection.java

```diff
@@ -40,4 +40,9 @@
      * have been handled, and all outgoing messages have been forwarded to the peer.
      */
     void stop();
+
+    /**
+     * Indicate that the execution containing this {@code ObjectConnection} has been prematurely stopped.
+     */
+    void stopNow();
```

Comment:
Can we call this `abort()`? Especially the field name `stoppedNow` is bugging me :)